src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -592,7 +597,15 @@ protected function initializeContainer()
         $this->container = require $cache->getPath();
         $this->container->set('kernel', $this);
 
-        if (!$fresh && $this->container->has('cache_warmer')) {
+        if ($fresh) {
+            return;
+        }
+
+        if ($oldContainer && get_class($this->container) !== $oldContainer->name) {
+            (new Filesystem())->remove(dirname($oldContainer->getFileName()));
```

Comment:
wouldn't this remove the whole cache dir ? This is not good, as it is not all about the DIC cache.